multipledispatch/dispatcher.py

```diff
@@ -260,11 +332,16 @@ def dispatch(self, *types):
             return None
 
     def dispatch_iter(self, *types):
+
         n = len(types)
         for signature in self.ordering:
             if len(signature) == n and all(map(issubclass, types, signature)):
                 result = self.funcs[signature]
                 yield result
+            elif len(signature) and isvariadic(signature[-1]):
```

Comment:
I believe this is guaranteed to be a `tuple`, in which case you can reduce to `elif signature and isvariadic(signature[-1]):`

You might even be able to reduce to just `isvariadic(signature[-1])` since I don't multiple dispatch allows an empty signature which means `signature[-1]` is always valid.

Reply:
I believe you can have an empty signature. This PR also adds tests that assert that explicitly empty supersedes a variadic.

Reply:
Interesting, it's definitely possible to define a function with an empty signature.